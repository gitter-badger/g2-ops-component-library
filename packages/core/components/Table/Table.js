// @flow

import type { Node } from 'react'
import React from 'react'

import {
  Table as MuiTable,
  TableBody,
  TableFooter,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn,
} from 'material-ui/Table'

import { wrapMuiContext } from '../../wrapMuiContext'

const Table = wrapMuiContext(MuiTable)

export { Table, TableBody, TableFooter, TableHeader, TableHeaderColumn, TableRow, TableRowColumn }
