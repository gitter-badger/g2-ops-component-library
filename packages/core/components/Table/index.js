export { Table, TableBody, TableFooter, TableHeader, TableHeaderColumn, TableRow, TableRowColumn } from './Table'
export default from './Table'
