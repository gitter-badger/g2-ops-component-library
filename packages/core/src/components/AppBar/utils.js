import { compose } from 'ramda'

const FIND_ALL_UNDERSCORES = /_+/g
const FIND_FIRST_LETTER_REGEX = /\w\S*/g

export const replaceUnderscoreWithSpace = (string) =>
  string && string.length > 0 && string.replace(FIND_ALL_UNDERSCORES, ' ')

export const capitalizeFirstLetter = (string) =>
  string &&
  string.replace(
    FIND_FIRST_LETTER_REGEX,
    (match) => match && `${match.charAt(0).toUpperCase()}${match.substr(1).toLowerCase()}`
  )

export const beautifyRoleText = compose(capitalizeFirstLetter, replaceUnderscoreWithSpace)