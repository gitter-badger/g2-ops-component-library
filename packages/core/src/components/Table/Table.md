Table Examples:

All the tables in Copart have to implement either RowSelectTable or Copart-Grid Components.

#### RowSelectTable:
* https://github.com/copartit/mui-row-select-table
* https://www.npmjs.com/package/mui-row-select-table

#### Copart-Grid:
* https://www.npmjs.com/package/copart-grid
* https://aldizh.github.io/copart-grid/