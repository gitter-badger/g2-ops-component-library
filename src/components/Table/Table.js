import React from 'react'
import {
  Table as MuiTable,
  TableBody,
  TableFooter,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn
} from 'material-ui/Table'
import { wrapMuiContext } from '../../wrapMuiContext'

/**
*
* @example ../../examples/Table.md
*/
const Table = (props) => (
  <MuiTable {...props}>
    {props.children}
  </MuiTable>
)

export default wrapMuiContext(Table)
export {
  TableBody,
  TableFooter,
  TableHeader,
  TableHeaderColumn,
  TableRow,
  TableRowColumn
}
