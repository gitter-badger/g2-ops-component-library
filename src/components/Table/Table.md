RowSelectTable:

All the tables in Copart have to implement either RowSelectTable Component.

For more information on accepted props refer to:
https://github.com/copartit/mui-row-select-table
https://www.npmjs.com/package/mui-row-select-table