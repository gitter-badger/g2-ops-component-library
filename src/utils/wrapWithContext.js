import React from 'react'
import WithContext from 'react-with-context'
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme'
import createMuiTheme from 'material-ui/styles/getMuiTheme'

const wrapWithContext = (component, context) => <WithContext context={context}>{component}</WithContext>

export const theme = createMuiTheme(baseTheme)

export const wrapWithMaterialUIContext = (component) =>
  wrapWithContext(component, { muiTheme: createMuiTheme(baseTheme) })

export default wrapWithContext
