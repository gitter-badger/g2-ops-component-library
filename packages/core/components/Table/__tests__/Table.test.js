import React from 'react'
import renderer from 'react-test-renderer'

import { wrapWithMaterialUIContext } from 'utils/wrapWithContext'

import Table, { TableHeader, TableHeaderColumn, TableBody, TableRow, TableRowColumn } from 'components/Table/Table'

describe('<Table />', () => {
  test('should render properly', () => {
    const tree = renderer.create(
      <Table>
        <TableHeader displaySelectAll={false}>
          <TableRow>
            <TableHeaderColumn>ID</TableHeaderColumn>
            <TableHeaderColumn>Name</TableHeaderColumn>
            <TableHeaderColumn>Status</TableHeaderColumn>
          </TableRow>
        </TableHeader>
        <TableBody displayRowCheckbox={false}>
          <TableRow>
            <TableRowColumn>1</TableRowColumn>
            <TableRowColumn>John Smith</TableRowColumn>
            <TableRowColumn>Employed</TableRowColumn>
          </TableRow>
          <TableRow>
            <TableRowColumn>2</TableRowColumn>
            <TableRowColumn>Randal White</TableRowColumn>
            <TableRowColumn>Unemployed</TableRowColumn>
          </TableRow>
          <TableRow>
            <TableRowColumn>3</TableRowColumn>
            <TableRowColumn>Stephanie Sanders</TableRowColumn>
            <TableRowColumn>Employed</TableRowColumn>
          </TableRow>
          <TableRow>
            <TableRowColumn>4</TableRowColumn>
            <TableRowColumn>Steve Brown</TableRowColumn>
            <TableRowColumn>Employed</TableRowColumn>
          </TableRow>
        </TableBody>
      </Table>,
    )
    expect(tree).toMatchSnapshot()
  })
})
