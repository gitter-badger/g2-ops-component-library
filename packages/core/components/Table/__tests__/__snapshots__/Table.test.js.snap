// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render properly 1`] = `
<div
  style={
    Object {
      "height": "auto",
      "muiPrepared": true,
      "overflow": "auto",
    }
  }
>
  <div
    style={
      Object {
        "muiPrepared": true,
      }
    }
  >
    <table
      className={undefined}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderCollapse": "collapse",
          "borderSpacing": 0,
          "fontFamily": "Roboto, sans-serif",
          "tableLayout": "fixed",
          "width": "100%",
        }
      }
    >
      <thead
        className={undefined}
        style={
          Object {
            "borderBottom": "1px solid #e0e0e0",
            "muiPrepared": true,
          }
        }
      >
        <tr
          className={undefined}
          style={
            Object {
              "borderBottom": "1px solid #e0e0e0",
              "color": "rgba(0, 0, 0, 0.87)",
              "height": 48,
              "muiPrepared": true,
            }
          }
        >
          <th
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "color": "#9e9e9e",
                "cursor": "default",
                "fontSize": 12,
                "fontWeight": "normal",
                "height": 56,
                "muiPrepared": true,
                "paddingLeft": 24,
                "paddingRight": 24,
                "position": "relative",
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
                "width": 24,
              }
            }
          />
          <th
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "color": "#9e9e9e",
                "fontSize": 12,
                "fontWeight": "normal",
                "height": 56,
                "muiPrepared": true,
                "paddingLeft": 24,
                "paddingRight": 24,
                "position": "relative",
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            ID
          </th>
          <th
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "color": "#9e9e9e",
                "fontSize": 12,
                "fontWeight": "normal",
                "height": 56,
                "muiPrepared": true,
                "paddingLeft": 24,
                "paddingRight": 24,
                "position": "relative",
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            Name
          </th>
          <th
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "color": "#9e9e9e",
                "fontSize": 12,
                "fontWeight": "normal",
                "height": 56,
                "muiPrepared": true,
                "paddingLeft": 24,
                "paddingRight": 24,
                "position": "relative",
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            Status
          </th>
        </tr>
      </thead>
    </table>
  </div>
  <div
    style={
      Object {
        "height": "inherit",
        "muiPrepared": true,
        "overflowX": "hidden",
        "overflowY": "auto",
      }
    }
  >
    <table
      className={undefined}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderCollapse": "collapse",
          "borderSpacing": 0,
          "fontFamily": "Roboto, sans-serif",
          "tableLayout": "fixed",
          "width": "100%",
        }
      }
    >
      <tbody
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <tr
          className={undefined}
          style={
            Object {
              "borderBottom": "1px solid #e0e0e0",
              "color": "rgba(0, 0, 0, 0.87)",
              "height": 48,
              "muiPrepared": true,
            }
          }
        >
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            1
          </td>
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            John Smith
          </td>
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            Employed
          </td>
        </tr>
        <tr
          className={undefined}
          style={
            Object {
              "borderBottom": "1px solid #e0e0e0",
              "color": "rgba(0, 0, 0, 0.87)",
              "height": 48,
              "muiPrepared": true,
            }
          }
        >
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            2
          </td>
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            Randal White
          </td>
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            Unemployed
          </td>
        </tr>
        <tr
          className={undefined}
          style={
            Object {
              "borderBottom": "1px solid #e0e0e0",
              "color": "rgba(0, 0, 0, 0.87)",
              "height": 48,
              "muiPrepared": true,
            }
          }
        >
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            3
          </td>
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            Stephanie Sanders
          </td>
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            Employed
          </td>
        </tr>
        <tr
          className={undefined}
          style={
            Object {
              "borderBottom": false,
              "color": "rgba(0, 0, 0, 0.87)",
              "height": 48,
              "muiPrepared": true,
            }
          }
        >
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            4
          </td>
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            Steve Brown
          </td>
          <td
            className={undefined}
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "backgroundColor": "inherit",
                "fontSize": 13,
                "height": 48,
                "muiPrepared": true,
                "overflow": "hidden",
                "paddingLeft": 24,
                "paddingRight": 24,
                "textAlign": "left",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            Employed
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
